plugins/scaffolder/src/next/TemplateWizardPage/Stepper/useTemplateSchema.ts

```diff
@@ -44,25 +48,40 @@ export const useTemplateSchema = (
       return stepFeatureFlag ? featureFlags.isActive(stepFeatureFlag) : true;
     })
     // Then filter out the properties that are not enabled with feature flag
-    .map(step => ({
-      ...step,
-      schema: {
-        ...step.schema,
-        // Title is rendered at the top of the page, so let's ignore this from jsonschemaform
-        title: undefined,
-        properties: Object.fromEntries(
-          Object.entries(step.schema.properties as JsonObject).filter(
-            ([key]) => {
-              const stepFeatureFlag =
-                step.uiSchema[key]?.['ui:backstage']?.featureFlag;
-              return stepFeatureFlag
-                ? featureFlags.isActive(stepFeatureFlag)
-                : true;
-            },
+    .map(step => {
+      const objectFieldTemplate = step.uiSchema['ui:ObjectFieldTemplate'] as
```

Comment:
Not sure if it makes sense to do this stuff in this hook. I'm thinking it makes more sense to do it somewhere in the `Stepper` component, or we write a hook to do this separately instead and call that hook in the stepper with the output of this hook maybe.

Reply:
I've created a `useTransformSchemaToProps` function that does this now.